tests/commands/train_test.py

```diff
@@ -38,6 +41,38 @@ def test_train_model(self):
 
         train_model(params, serialization_dir=os.path.join(self.TEST_DIR, 'test_train_model'))
 
+    def test_error_is_throw_when_cuda_device_is_not_available(self):
+        params = Params({
+                "model": {
+                        "type": "simple_tagger",
+                        "text_field_embedder": {
+                                "tokens": {
+                                        "type": "embedding",
+                                        "embedding_dim": 5
+                                }
+                        },
+                        "encoder": {
+                                "type": "lstm",
+                                "input_size": 5,
+                                "hidden_size": 7,
+                                "num_layers": 2
+                        }
+                },
+                "dataset_reader": {"type": "sequence_tagging"},
+                "train_data_path": 'tests/fixtures/data/sequence_tagging.tsv',
+                "validation_data_path": 'tests/fixtures/data/sequence_tagging.tsv',
+                "iterator": {"type": "basic", "batch_size": 2},
+                "trainer": {
+                        "num_epochs": 2,
+                        "cuda_device": torch.cuda.device_count(),
+                        "optimizer": "adam"
+                }
+        })
+
+        with self.assertRaises(ConfigurationError) as context:
+            train_model(params, serialization_dir=os.path.join(self.TEST_DIR, 'test_train_model'))
+        self.assertTrue('specified a GPU but none is available' in str(context.exception))
```

Comment:
for consistency with the rest of the codebase (and because I think it's cleaner), could you change this test to something like

```
with pytest.raises(ConfigurationError, message="..."):
   ...
```